autopush-common/src/reliability.rs

```diff
@@ -106,14 +118,15 @@ impl PushReliability {
                     .unwrap_or_else(|| "None".to_owned()),
                 new
             );
-            if let Ok(mut conn) = client.get_connection_with_timeout(CONNECTION_EXPIRATION) {
-                self.internal_record(&mut conn, old, new, expr, id).await;
-            }
+            match client.get_connection_with_timeout(CONNECTION_EXPIRATION) {
+                Ok(mut conn) => self.internal_record(&mut conn, old, new, expr, id).await,
+                Err(e) => warn!("🔍⚠️ Unable to record reliability state: {:?}", e),
+            };
```

Comment:
nit: I'd encourage `inspect_err` for this one here especially as it's used elsewhere in this module

Reply:
I'm going to argue that one, only because the `Ok` does require action that the error case cannot provide. If there's an error, we do _not_ want to call _internal_record()`.